src/components/tickets/useTableSort.ts

```diff
@@ -0,0 +1,56 @@
+import { fetchTicketAsync } from '../../store/tickets/thunk';
+import { useAppDispatch } from '../../store/hooks';
+import { useSearchParams } from 'react-router-dom';
+import { useState } from 'react';
+interface IProps {
+  searchParams: URLSearchParams;
+  fetchTickets: () => Promise<void>;
+  selectedPriorities: string[];
+  setSelectedPriorities: React.Dispatch<React.SetStateAction<string[]>>;
+  paginationModel: {
+    pageSize: number;
+    page: number;
+  };
+  setPaginationModel: React.Dispatch<
+    React.SetStateAction<{
+      pageSize: number;
+      page: number;
+    }>
+  >;
+}
+export const useTableSortAndFilter = (): IProps => {
+  const dispatch = useAppDispatch();
+
+  const [searchParams, setSearchParams] = useSearchParams();
+  const priority = searchParams.getAll('priority');
+  const _page = searchParams.get('_page') ?? '0';
+  const _limit = searchParams.get('_limit') ?? '8';
+
+  const [selectedPriorities, setSelectedPriorities] = useState<string[]>(priority);
+  const [paginationModel, setPaginationModel] = useState({
+    pageSize: +_limit,
+    page: _page === '0' ? +_page : +_page - 1,
+  });
+
+  const fetchTickets = async () => {
+    await setSearchParams((params) => {
```

Comment:
What if we already have `_page` and `_limit` params?

Reply:
If we already have params then we will get them in PaginationModel and then we take them from PaginationModel here  - so they will be reset to the params we already have.